src/io/hdf/hdf_archive.h

```diff
@@ -89,16 +89,22 @@ class hdf_archive
   template<class Comm = Communicate*>
   hdf_archive(Comm c, bool request_pio = false) : file_id(is_closed), access_id(H5P_DEFAULT), xfer_plist(H5P_DEFAULT)
   {
-    H5Eget_auto2(H5E_DEFAULT, &err_func, &client_data);
-    H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
-    set_access_plist(c, request_pio);
+#pragma omp critical
```

Comment:
The purpose of the replacing the error functions is to avoid extraneous and often confusing messages from HDF.  It looks like the destructor is supposed to replace the error-handling function captured here.  However, that won't work as intended if multiple threads call this function - only the first thread will store the real error handling function, the rest will get NULL.   This also happens with multiple hdf_archives in use - only the first one will capture the original error-handling function.
It seems like maybe the error suppression in HDF should be handled differently (not for this PR, though) - maybe just turn it off at the beginning of the run and don't bother turning back on?

Reply:
Moved this to issue #4283